services/src/main/java/org/keycloak/services/util/CookieHelper.java

```diff
@@ -128,13 +125,21 @@ public static void addCookie(String name, String value, String path, String doma
     }
 
 
-    public static Set<String> parseCookie(String cookieHeader, String name) {
-        String parts[] = cookieHeader.split("[;,]");
+    public static Set<String> parseCookie(String header, String name) {
```

Comment:
Parsing cookies like that every time we look up a specific cookie is not optimal. Ideally, we should be parsing only once. 

Also, I'm not sure what motivated parsing the `Cookie` header is still valid. See https://issues.redhat.com/browse/KEYCLOAK-5270.

Reply:
You mean removing the part where cookies are parsed from a request's header and keep only this ` Cookie cookie = headers.getCookies().get(name);`?

If we still want to or need to support IE <= 11 we need to keep it otherwise [KEYCLOAK-5270](https://issues.redhat.com/browse/KEYCLOAK-5270) will become again an issue. I agree it's not nice but it's a workaround for a bug in IE which doesn't handle cookies with overlapping paths correctly.

Reply:
IE 10 is supposed to not get any updates including security fixes. I'm not sure if we want to support a browser in this situation. Thus my question.

IE 11 is still there I think but will soon enter EOL too.

Reply:
@martin-kanis But yeah, if we want to be more defensive than we can change the current code to correctly follow the spec.

Reply:
IE11 is unfortunately still supported https://access.redhat.com/articles/2342861 :( It seems we need to live with historic browsers and rather not introduce regressions as you know how the browser-specific regressions are painful to debug and resolve (As it is the end-user with IE11 for whom the things stopped working and developers/administrators usually don't have chance to test it on their own due the fact that they don't have easy access to those prehistoric browsers)